rfcs/0000-question-sets.md

```diff
@@ -0,0 +1,150 @@
+# Summary
+
+A mechanism for allowing sets of questions to be published and used by multiple courses.
```

Comment:
I guess I'm wondering whether we should call these "sharable courses" by a different name ("questionSets") or whether we should just allow any course to set a flag like `"sharable": true`?

One advantage of making full courses sharable is that they could include not just lists of questions, but also template assessments that could be copied. For example, if I made a bank of questions for "Learning Python 101", I could also put together sample homeworks (and even exams). We could have a "copy assessment" button that mimics the current "copy question" button, so that another instructor could copy the homeworks into their course and the questions would all be automatically linked back to the source course. Then they'd just need to adjust the dates, maybe add or delete a question or two, and they'd be ready to go.

I could even provide entire sample course instances that could be copied. Maybe I make course instances for a "Full 15-week Python Course" and another "4-week intro course". Then we could have a button so an instructor could just copy the entire course instance, and then go through and make small tweaks as they want (editing dates, at a minimum).

I think this "course in a box" could be very powerful.

Reply:
I like this idea! I'd still want the same semantics + restrictions as outlined below for questions in a shareable course, but that wouldn't impact our ability to provide assessments like you're suggesting.

Reply:
Yep! I like all those restrictions below for sharable courses.

Reply:
> whether we should just allow any course to set a flag like `"sharable": true`?

I'm still not too familiar with PL access rules, so apologies if this was actually addressed in the Access Control section (course access roles?), but:

If an instructor sets `sharable`, they probably won't want it to just be open season for anyone to click "clone question" (esp other schools). This has come up with MCS/Coursera, where instructors seemed fine with sharing their content with other schools but only on their own terms.

Reply:
Oh yes! Totally agreed! This PR doesn't describe anything about how sharing permissions work (although maybe this PR would be a good place to add this?). In our earlier conversations we had talked about having explicit permissions (like "Course X" can use questions from "Course Y") which would be set by the course owner.